shopfloor/actions/stock.py

```diff
@@ -10,6 +13,96 @@ class StockAction(Component):
+        return_type = return_picking.picking_type_id
+        return {
+            "product_id": product.id,
+            "product_uom": product.uom_id.id,
+            "picking_id": return_picking.id,
+            "state": "draft",
+            "date": fields.Datetime.now(),
+            "location_id": return_picking.location_id.id,
+            "location_dest_id": return_picking.location_dest_id.id,
+            "picking_type_id": return_type.id,
+            "warehouse_id": return_type.warehouse_id.id,
+            "origin_returned_move_id": origin_move.id,
+            "procure_method": "make_to_stock",
+        }
+
+    def _create_return_move__link_to_origin(self, return_move, origin_move):
+        move_orig_to_link = origin_move.move_dest_ids.mapped("returned_move_ids")
+        move_orig_to_link |= origin_move
+        origin_move_dest = origin_move.move_dest_ids.filtered(
+            lambda m: m.state not in ("cancel")
+        )
+        move_orig_to_link |= origin_move_dest.move_orig_ids.filtered(
+            lambda m: m.state not in ("cancel")
+        )
+        move_dest_to_link = origin_move.move_orig_ids.mapped("returned_move_ids")
+        move_dest_orig = origin_move.returned_move_ids.move_orig_ids.filtered(
+            lambda m: m.state not in ("cancel")
+        )
+        move_dest_to_link |= move_dest_orig.move_dest_ids.filtered(
+            lambda m: m.state not in ("cancel")
+        )
+        write_vals = {
+            "move_orig_ids": [(4, m.id) for m in move_orig_to_link],
+            "move_dest_ids": [(4, m.id) for m in move_dest_to_link],
+        }
+        return_move.write(write_vals)
+
+    def create_return_move(self, return_picking, origin_moves):
+        """Creates a return move for a given return picking / move"""
+        # Logic has been copied from
+        # odoo_src/addons/stock/wizard/stock_picking_return.py
+        for origin_move in origin_moves:
+            # If max qty <= 0, it means that everything has been returned already.
+            # Try with the next one from the recordset.
+            max_qty = self._create_return_move__get_max_qty(origin_move)
+            if max_qty > 0:
+                return_move_vals = self._create_return_move__get_vals(
+                    return_picking, origin_move
+                )
+                return_move_vals.update(product_uom_qty=max_qty)
+                return_move = origin_move.copy(return_move_vals)
+                self._create_return_move__link_to_origin(return_move, origin_move)
+                return return_move
+
+    def _create_return_picking__get_vals(self, return_types, origin):
+        return_type = fields.first(return_types)
+        return {
+            "move_lines": [],
+            "picking_type_id": return_type.id,
+            "state": "draft",
+            "origin": origin,
+            "location_id": return_type.default_location_src_id.id,
+            "location_dest_id": return_type.default_location_dest_id.id,
```

Comment:
The location on the picking type could be empty and this will cause the new picking creation to fail with an error 500. Add a check that the return picking is properly configured otherwise raise a proper error